pkg/proxy/util/utils_linux.go

```diff
@@ -0,0 +1,73 @@
+// +build linux
+
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package util
+
+import (
+	"fmt"
+	"net"
+	"syscall"
+
+	"github.com/vishvananda/netlink"
+)
+
+type NetlinkHandle interface {
+	AddrAdd(link netlink.Link, addr *netlink.Addr) error
+	AddrDel(link netlink.Link, addr *netlink.Addr) error
+	LinkByName(dev string) (netlink.Link, error)
+}
+
+func NewNetLinkHandle() NetlinkHandle {
+	return &netlink.Handle{}
+}
+
+// EnsureAddressBind checks if address is bound to the interface and, if not, binds it. If the address is already bound, return true.
+func EnsureAddressBind(address, devName string, h NetlinkHandle) (exist bool, err error) {
```

Comment:
This method looks odd, why do you take an interface as a parameter?

What about

```
type NetlinkHandle interface {
 	EnsureAddressBind(address, devName string)
 }
```

?

Reply:
Good suggestion.